### PR TITLE
fix: parallelism fix for ifname processor

### LINKED ISSUE #9983: ifName plugin parallel lookups fails
It seems like the `processors.ifname` setting `max_parallel_lookups` isn't working as expected. When I added more devices, the plugin started failing to get the ifname for some of them.

### Relevant telegraf.conf:
Used default `processors.ifname` config (`max_parallel_lookups = 100`)

### System info:

Telegraf 1.20.2 (git: HEAD f721f53d)

### Steps to reproduce:

1. add lot of metrics from different agents 

### Expected behavior:

ifname processor to perfectly be able to do lookups in parallel.

### Actual behavior:

Randomly getting these kind of errors:

> D! [processors.ifname] Error adding tag interface number 126 isn't in the table of interface names

### Additional info:

Although there seem to be signals in place to only do one lookup per agent, the lookup function itself does not seem to be thread-safe implemented: https://github.com/influxdata/telegraf/blob/8265969f523cddb601211e4f070fef4a56b7aae0/plugins/processors/ifname/ifname.go#L301-L312

### PR DESCRIPTION
**Pull request by Hipska**
### Required for all PRs:

- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #9983

This PR aims to fix thread-safety for the ifname plugin. Before it used only one instance of gosnmp for all agents, but that doesn't work out if multiple agents need to be processed at the same time. My addition creates a gosnmp instance for each agent, so no threading problems occur.

### CODE REVIEW

File: plugins/processors/ifname/ifname.go [hunk 1]
```diff
@@ -299,7 +296,7 @@ func (d *IfName) getMap(agent string) (entry nameMap, age time.Duration, err err
 }
 
 func (d *IfName) getMapRemoteNoMock(agent string) (nameMap, error) {
-	gs := d.gsBase
+	gs, _ := snmp.NewWrapper(d.ClientConfig)
```
[the remark this answers is not in the document]
Comment (Hipska) [reply plugins/processors/ifname/ifname.go:294]: @reimda What do you think?

[the remark this answers is not in the document]
Comment (Hipska) [reply plugins/processors/ifname/ifname.go]: Nevermind, You are right, I added the check..

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @Hipska!

**Review by sspaink — APPROVED** [review]

**Comment by Hipska** [comment]
Got a follow-up ready as well: #10214

**Pull request merged**